### PR TITLE
Geirst/rewrite enum store dictionary

### PR DESCRIPTION
**Pull request by geirst**
@toregge please review

### CODE REVIEW

File: vespalib/src/vespa/vespalib/datastore/unique_store_dictionary.h [hunk 1]
```diff
@@ -9,20 +9,17 @@ namespace search::datastore {
 
 class EntryComparatorWrapper;
 
-/*
+/**
  * A dictionary for unique store. Mostly accessed via base class.
  */
+template <typename DictionaryType>
```
Comment (toregge) [inline vespalib/src/vespa/vespalib/datastore/unique_store_dictionary.h]: Consider another template argument name, e.g. DictionaryT

File: vespalib/src/vespa/vespalib/datastore/unique_store_dictionary.h [hunk 2]
```diff
@@ -9,20 +9,17 @@ namespace search::datastore {
...
 class UniqueStoreDictionary : public UniqueStoreDictionaryBase
 {
 public:
-    using DictionaryTraits = btree::BTreeTraits<32, 32, 7, true>;
-    using Dictionary = btree::BTree<EntryRef, btree::BTreeNoLeafData,
-                                    btree::NoAggregated,
-                                    EntryComparatorWrapper,
-                                    DictionaryTraits>;
+    using Dictionary = DictionaryType;
```
Comment (toregge) [inline vespalib/src/vespa/vespalib/datastore/unique_store_dictionary.h]: Consider using DictionaryType = DictionaryT

**Review by toregge — APPROVED** [review]

File: vespalib/src/vespa/vespalib/datastore/unique_store_dictionary.h (hunk 1 above)
Comment (geirst) [reply vespalib/src/vespa/vespalib/datastore/unique_store_dictionary.h]: Fixed

File: vespalib/src/vespa/vespalib/datastore/unique_store_dictionary.h (hunk 2 above)
Comment (geirst) [reply vespalib/src/vespa/vespalib/datastore/unique_store_dictionary.h]: Fixed

**Pull request merged**